app/api/common.v2/database/MongoDataSource.ts

```diff
@@ -16,17 +16,32 @@ export abstract class MongoDataSource<CollectionSchema extends Document = any> {
     this.transactionManager = transactionManager;
   }
 
-  protected getCollection() {
+  protected getCollection(collectionName = this.collectionName) {
     return new SyncedCollection<CollectionSchema>(
       new SessionScopedCollection<CollectionSchema>(
-        this.db.collection(this.collectionName),
+        this.db.collection(collectionName),
         this.transactionManager
       ),
       this.transactionManager,
       this.db
     );
   }
 
+  protected async collectionExists(): Promise<boolean> {
+    const collections = await this.db.listCollections({ name: this.collectionName }).toArray();
+    return collections.length > 0;
+  }
+
+  protected async dropCollection() {
+    if (await this.collectionExists()) {
+      await this.db.dropCollection(this.collectionName, { session: this.getSession() });
+    }
+  }
+
+  protected async createCollection() {
+    await this.db.createCollection(this.collectionName, { session: this.getSession() });
```

Comment:
This piece of code was extracted exactly as it was in it's original place. If I'm correct it will fail when trying to create a an already existing collection. I don't think it's a bad thing if it fails because making it 'non-failing' should be a responsibility of the consuming code. That brings me to another line of thinking which is: we should not add the check here, but actually move the one in the drop back to the consumer.